clientv3/watch.go

```diff
@@ -251,7 +254,7 @@ func (w *watcher) run() {
 		// New events from the watch client
 		case pbresp := <-w.respc:
 			switch {
-			case pbresp.Canceled:
+			case !pbresp.Created && pbresp.Canceled:
 				delete(cancelSet, pbresp.WatchId)
 			case pbresp.Created:
```

Comment:
can we move pbresp.Created to the top? then we do not need !pbresp.Created checking.